### PR TITLE
[Serve] Use dashboard head in Serve CLI

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

This change makes the Serve CLI use the dashboard head instead of the dashboard agent.

## Related issue number

Addresses #40048

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - This change relies on existing tests.

### CODE REVIEW

File: python/ray/serve/scripts.py [hunk 1]
```diff
@@ -128,6 +128,15 @@ def convert_args_to_dict(args: Tuple[str]) -> Dict[str, str]:
     return args_dict
 
 
+def warn_if_agent_address_set():
+    if "RAY_AGENT_ADDRESS" in os.environ:
+        cli_logger.warning(
+            "The `RAY_AGENT_ADDRESS` env var has been deprecated in favor of "
+            "the `RAY_HEAD_ADDRESS` env var. The `RAY_AGENT_ADDRESS` is "
```
Comment (edoakes) [inline python/ray/serve/scripts.py:135]: I think you mean `RAY_DASHBOARD_ADDRESS`?

**Review by sihanwang41 — APPROVED** [review]
LGTM with fixing Ed's comments.

File: python/ray/serve/scripts.py (hunk 1 above)
Comment (shrekris-anyscale) [reply python/ray/serve/scripts.py]: Yeah, thanks for the catch!

**Review by edoakes — APPROVED** [review]

**Comment by shrekris-anyscale** [comment]
The test failures are unrelated. @edoakes this change is ready to merge.

**Pull request merged**